examples/BERT/metrics.py

```diff
@@ -0,0 +1,72 @@
+import collections
+import re
+import string
+
+
+def compute_qa_exact(ans_pred_tokens_samples):
+
+    '''
+        Input: ans_pred_tokens_samples: [([ans1_tokens_candidate1, ans1_tokens_candidate2], pred1_tokens),
+                                         ([ans2_tokens_candidate1, ans2_tokens_candidate2], pred2_tokens),
+                                         ...
+                                         ([ansn_tokens_candidate1, ansn_tokens_candidate2], predn_tokens)]
+        ans1_tokens_candidate1 = ['this', 'is', 'an', 'sample', 'example']
+        Output: exact score of the samples
+    '''
+
+    def normalize_txt(text):
+        # lower case
+        text = text.lower()
+
+        # remove punc
+        exclude = set(string.punctuation)
+        text = "".join(ch for ch in text if ch not in exclude)
+
+        # remove articles
+        regex = re.compile(r"\b(a|an|the)\b", re.UNICODE)
+        text = re.sub(regex, " ", text)
+
+        # white space fix
+        return " ".join(text.split())
+
+    exact_scores = []
+    for (ans_tokens, pred_tokens) in ans_pred_tokens_samples:
+        pred_str = " ".join(pred_tokens)
+        candidate_score = []
+        for item in ans_tokens:
+            ans_str = " ".join(item)
+            candidate_score.append(int(normalize_txt(ans_str) == normalize_txt(pred_str)))
+        exact_scores.append(max(candidate_score))
+    return 100.0 * sum(exact_scores) / len(exact_scores)
+
+
+def compute_qa_f1(ans_pred_tokens_samples):
```

Comment:
can this be turned into a generic f1 metric that we can throw into torchtext?

Reply:
`sample_f1` func can be landed in torchtext.